doc/ref_arch/kubernetes/chapters/chapter03.md

```diff
@@ -59,13 +59,31 @@
 <a name="3.2.2"></a>
 ### 3.2.2 Container Networking Services
 
-> This chapter should discuss networking related considerations of container network services, like:
-> * Ingress and any limitations on protocol or raw interface support
-> * Dependencies on external loadbalancers
-> * Regular service mesh related considerations
-> * Network Service Mesh related considerations
-> * Number of interfaces and IP addresses to be supported by a pod
-> * IPv6 single and dual stack needs
+Becouse `req.inf.ntw.01` requires the architecture to support CNI and `req.inf.ntw.16` requires the capability to attach several network interfaces to the pods the architecture must suport a CNI metaplugin/CNI multiplexer.
+
+A CNI metaplugin/CNI multiplexer is capable to attach several interfaces, using different other CNI plugins to a pod. The different network charastheristics of the interfaces, which require different networking technologies require different CNI plugins.
+
+Inter node communication required by `req.inf.ntw.02` must be served by a simple CNI using overlay technology, like Flannel.
+
+There are two types of low latency and high throughput networks required by `req.inf.ntw.04`. Network used for signalling traffic are more demanding than what an overlay network can handle, but still does not need the usage of user space networking. Due to the nature of signalling protocols used this type of netorks require the NAT-less communication stated by `req.inf.ntw.03`. Due to the combination of these two requirements networks with this charactheristics must be served by a CNI plugin with IPVLAN or MACVLAN suport.
+
+The low latency, high throughput networks for handling the user plane traffic require the capability to use an user space networking technology, like VPP or DPDK. In case of DPDK the usage of SR-IOV VF-s is recommended.
```

Comment:
SR-IOV - do we need to specify that we don't recommend PCI Pass Through?  (not my strong point, simply parroting what I hear @rabi-abdel shouting into the phone every day 😉)

Reply:
I don't know that those words are being used to mean what they're supposed to mean (the reason being they're abused on a daily basis, so it's not terribly much a surprise).

SR-IOV: how we split one PCI device into multiple devices, which is an option

PCI passthrough: in the container context, handing a PCI device into a process's direct control.  It may be an SR-IOV derived device.  It might not be.  It might be a device that's not capable of SR-IOV.

It is likely a very bad idea to hand an CNF control of the PF of a device that *does* support SR-IOV, because that gives it the power to make new PCI devices - which will end up with the kernel, and that's almost certainly a really bad idea.  On the other hand, it might make sense to hand a trusted service control of that device so that it can manipulate the VFs for SDN purposes.  I believe (and I think Trevor Cooper knows more about this than me) that the DPDK libraries and tools support controlling PFs in this way.

Also, diverging from the networking discussion for a moment, whatever means of providing passthrough devices we choose might want pulling out from the exclusively networking section since it's also a reasonable thing to do with accelerators and presenting the PCI devices to the container consistently and in a way that's passthrough-friendly should be the same, for usability reasons.

Finally, I would take the comment about the way those devices are used as an instruction to presentation.  The models we see in the wild today tend to stick even PCI NICs into the kernel netns as network devices and they have to be removed from the netns to be consumed by a DPDK application and rebound to a passthrough kernel device driver.  It would be better if we forgot they were networking devices at all and presented them to the container as a vfio device file, since that device can be used with no privilege to the consuming container.

Reply:
Re ... _It would be better if we forgot they were networking devices at all and presented them to the container as a vfio device file, since that device can be used with no privilege to the consuming container._

Agree.

Reply:
_In case of DPDK the usage of SR-IOV VF-s is recommended_
we need to clarify what that means. if it 100% OK to use SR-IOV to present a NIC as multiple NICs, but it is not ok to expose that VF to container without any abstraction via a standard interface (memif, others ...) we need to pick a way.

Reply:
ohmm, bifurcated drivers? using SR-IOV devices for kernel networking purposes?
yes, with _Intel_ NICs for _DPDK_ purposes what you write is true, but it is hardly the only use-case

you keep saying it is not okay to expose VFs for containers through the Devices abstraction layer, but I still haven't seen the exact reasons why

Reply:
Can we conclude this debate? @rabi-abdel any responses to @Levovar -s comments?

Reply:
As we agreed on the [2019-12-19 - CNTT - RA2 - All call](https://wiki.lfnetworking.org/display/LN/2019-12-19+-+%5BCNTT+-+RA2+-+All%5D+-+Meeting+Agenda+and+Minutes) I will remove the recommendation for SR-IOV with DPDK. I will also add a note that it is still under discussion along with an other note stating that an infra can still offer the possibility without being CNTT un compliant.